### DISCUSSION TITLE
is `compactVerify` supposed to throw JWTExpired?

### DISCUSSION THREAD

**Discussion by jayenashar** [panva/jose#161] [Q&A] [answered]
I'm looking at https://github.com/panva/jose/discussions/62 and it seems like it should throw an error if the token is expired, but it's not happenning for me and i have to check expiration explicitly.  Is there some argument that will get it to check exp and nbf?

```
  let userId;
  try {
    const {payload} = await compactVerify(authToken, refreshTokenPublicKey);

    var obj = JSON.parse(decoder.decode(payload));
    userId = obj.sub; // sub is user id in refresh token

    if (Date.now() >= obj.exp * 1000) {
      res.status(401).json({message: 'Unauthorized! Expired credentials'});  
      return;
    }

  } catch(error) {
    res.status(401).json({message: 'Unauthorized! Invalid credentials'});
    return;
  }
```

**Comment by panva** [comment, marked as answer]
> is `compactVerify` supposed to throw JWTExpired?

No. compactVerify is not processing the payload of a token at all, it's a generic JWS utility. Maybe you should use the module targetted at validating JWTs instead of one purely for JWS.

[jose/jwt/verify](https://github.com/panva/jose/blob/main/docs/functions/_jwt_verify_.jwtverify.md#readme) (Verifies the JWT format (to be a JWS Compact format), verifies the JWS signature, validates the JWT Claims Set.) not [jose/jws/compact/verify](https://github.com/panva/jose/blob/main/docs/functions/_jws_compact_verify_.compactverify.md#readme) (Verifies the signature and format of and afterwards decodes the Compact JWS.)

**Reply by jayenashar** [reply]
thank you!
